transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -147,8 +148,23 @@ private void writeTarget()
             }
             catch ( final IOException e )
             {
-                throw new TransferException( "Failed to write to local proxy store: {}\nOriginal URL: {}. Reason: {}",
-                                             e, target, url, e.getMessage() );
+                ConcreteResource resource = null;
+                if ( target.exists() )
```

Comment:
I think we can avoid checking if it exists and just delete it. But first, you'll have to use IOUtils.closeQuietly(out) to ensure the file can be deleted.

Reply:
Yes.
These code in catch block,In finally block we have IOUtils.closeQuietly(out) , I think finally always 
execute firstly. Is it good enough?

Reply:
This is the opposite of my understanding of how finally works. Can you please verify this before we rely on that effect? 

In any case, IMO it improves clarity of the code to have the extra `closeQuietly()` call in the catch block, and it will not have a severe impact on performance.

Reply:
Sorry, My fault , finally block just before throw exception and break the method.
Honestly ,I never write logic in catch block, so I'm misunderstanding the basic concept. 
I will using it.